client/src/main/java/io/pravega/client/segment/impl/EventSegmentReader.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.segment.impl;
+
+import io.pravega.client.stream.EventStreamWriter;
+import java.nio.ByteBuffer;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Defines a InputStream for a single segment.
+ * Once created the offset must be provided by calling setOffset.
+ * The next read will proceed from this offset. Subsequent reads will read from where the previous
+ * read call on this instance left off. (Parallel calls to read data will be serialized)
+ * Get offset can be used to store a location to revert back to that position in the future.
+ */
+public interface EventSegmentReader extends AutoCloseable {
+    
+    Segment getSegmentId();
+    
+    /**
+     * Sets the offset for reading from the segment.
+     *
+     * @param offset The offset to set.
+     */
+    public abstract void setOffset(long offset);
+
+    /**
+     * Gets the current offset. (Passing this to setOffset in the future will reset reads to the
+     * current position in the segment.)
+     *
+     * @return The current offset.
+     */
+    public abstract long getOffset();
+
+    /**
+     * Reads a single event's bytes from the segment. Buffering is performed internally to try to prevent
```

Comment:
`Reads a single event's bytes from the segment.` -> `Reads the bytes of a single event from the segment.`